worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala

```diff
@@ -64,27 +64,29 @@ abstract private[worker] class Flusher(
         override def run(): Unit = {
           while (!stopFlag.get()) {
             val task = workingQueues(index).take()
-            val key = s"Flusher-$this-${Random.nextInt()}"
-            workerSource.sample(WorkerSource.FLUSH_DATA_TIME, key) {
-              if (!task.notifier.hasException) {
-                try {
+            try {
+              val key = s"Flusher-$this-${Random.nextInt()}"
+              workerSource.sample(WorkerSource.FLUSH_DATA_TIME, key) {
+                if (!task.notifier.hasException) {
                   val flushBeginTime = System.nanoTime()
                   lastBeginFlushTime.set(index, flushBeginTime)
                   task.flush()
                   if (flushTimeMetric != null) {
                     val delta = System.nanoTime() - flushBeginTime
                     flushTimeMetric.update(delta)
                   }
-                } catch {
-                  case _: ClosedByInterruptException =>
-                  case e: IOException =>
-                    task.notifier.setException(e)
-                    processIOException(e, DiskStatus.READ_OR_WRITE_FAILURE)
                 }
                 lastBeginFlushTime.set(index, -1)
               }
               returnBuffer(task.buffer)
               task.notifier.numPendingFlushes.decrementAndGet()
+            } catch {
+              case e: Throwable =>
```

Comment:
moving the exception-catching block silently causes behavior change: line 84-87 skipped to execute when IOE throws now.